language-plutus-core/test/types/recursiveUnwrap.plc

```diff
@@ -1,18 +0,0 @@
-(program 1.0.0
```

Comment:
Why did you delete this test?

Reply:
It's not valid according to the spec.

Reply:
Why not fix it? Reducing our test coverage is bad.

Reply:
It's not obvious to me how to do so, but I will have a look at fixing it.

Reply:
Looks valid to me - what's wrong with it?

Reply:
This is confusing, since it actually involves an old version of the spec. In the spec this repo is currently tied to, a `(lam x ty t)` must have `ty` be a type value. Thus, the 

```
(fix list (all r (type) (fun r (fun (fun [(con integer) (con 64)] (fun list r)) r))))
```

type annotation is not valid.

Reply:
Hm, I see. In the current version there's a workaround to avoid the fact that we can't reduce builtin applications for terms - the special "builtin value" rule allows us to have unreduced builtin applications in values. Sadly we're missing the corresponding thing at the type level. AFAIC that means that it's impossible to abstract over types containing builtin types atm. Good thing we're getting rid of type values!

However, an easy fix suggests itself for the test: simply parameterize the function by a type and a value to use in the function. That should then work fine. It's a bit more verbose than the current version using concrete types, but oh well.

Reply:
That seems sensible to me. I will try to do this.